### PR TITLE
Fix deprecations and warnings

### LINKED ISSUE #167: Ruby 2.7 warning about PublicSuffix::List.default
Seen in testing:
```
.../publicsuffix-ruby/lib/public_suffix/list.rb:51: warning: Using the last argument as keyword parameters is deprecated; maybe ** should be added to the call
.../publicsuffix-ruby/lib/public_suffix/list.rb:69: warning: The called method `parse' is defined here
```

### PR DESCRIPTION
**Pull request by BrianHawley**
[Fixes #167]

### CODE REVIEW

File: test/test_helper.rb [hunk 1]
```diff
@@ -10,7 +10,7 @@
 
 require "minitest/autorun"
 require "minitest/reporters"
-require "mocha/setup"
+require "mocha/minitest"
```
Comment (weppos) [inline test/test_helper.rb]: Is this change directly related to the warning issues?

File: .gitignore [hunk 2]
```diff
@@ -9,3 +9,7 @@ pkg/*
 # YARD
 .yardoc
 yardoc/
+
```
Comment (weppos) [inline .gitignore]: Can you please revert these changes? They are not relevant for the fix itself.

File: test/test_helper.rb (hunk 1 above)
Comment (BrianHawley) [reply test/test_helper.rb]: It's related to a separate warning that came up while I was running the test suite locally. The mocha gem tells you that "mocha/setup" is deprecated, and that you should require one of the more specific setups. This is the appropriate one for your test suite.

File: .gitignore (hunk 2 above)
Comment (BrianHawley) [reply .gitignore]: Sure. It would be good if you readded it though.

**Pull request merged**